docs/guides/initializers.md

```diff
@@ -0,0 +1,123 @@
+---
+layer = nn.Dense(features=5)
+default_params = layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params']
+
+layer = nn.Dense(features=5, kernel_init=lecun_normal(), bias_init=zeros)
+manual_params = layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params']
+
+# should get the same param values
+assert (default_params['kernel'] == manual_params['kernel']).all()
+assert (default_params['bias'] == manual_params['bias']).all()
+```
+
++++ {"id": "jPW_E1ZqHdrQ"}
+
+## `Initializer` function signature
+
++++ {"id": "3Kglqd9vuxTG"}
+
+To maintain consistency, all `Initializer` functions that are passed to the `kernel_init` and `bias_init` arguments **must follow the function signature: `[PRNGKey, Shape, Dtype] -> Array`**. Most functions in the [Flax initializer list](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#module-flax.linen.initializers) are **builder functions** and build an `Initializer` function that follows this function signature. The two exceptions are [`flax.linen.initializers.zeros`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.zeros.html) and [`flax.linen.initializers.ones`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.ones.html), which are already `Initializer` functions that follow the function signature. This is why in the above example, we must call `lecun_normal()` to build an `Initializer` function, whereas we can directly use `zeros` since it's already an `Initializer` function.
+
++++ {"id": "sG5oygwB7UrW"}
+
+Note that even if the `Initializer` is deterministic, **a PRNGKey must be passed**. Therefore functions like [`jax.numpy.zeros`](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.zeros.html) and [`jax.numpy.ones`](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ones.html) can't be used as initializers as they have an invalid function signature.
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 60
+  status: ok
+  timestamp: 1673133041047
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: IjLW1abzw2pM
+outputId: 2616d907-7e24-4cd3-e926-3c39c8dd5401
+---
+layer = nn.Dense(features=5, kernel_init=jnp.zeros) # jnp.zeros has an invalid function signature
+try:
+  layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params'] # <-- throws an error
+except Exception as e:
+  print(f'Caught error: {e}')
+```
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 62
+  status: ok
+  timestamp: 1673133041242
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: RlOIOCj68p86
+outputId: 5871f462-1d24-47f9-cc20-6fa7f358ea99
+---
+layer = nn.Dense(features=5, kernel_init=zeros) # use flax.linen.initializers.zeros instead
+layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params'] # <-- successfully generate params
+```
+
++++ {"id": "S4X_xHHk-b4V"}
+
+## `Initializer` restrictions for `bias_init`
```

Comment:
Nice! Maybe we should link to this from our `bias_init` docstrings as well?

Reply:
Do you mean all docstrings that contain a description for `bias_init` (like [here](https://github.com/google/flax/blob/main/flax/linen/linear.py#L167))? Something like: `bias_init: initializer function for the bias. To see restrictions on valid initializers, refer to our [guide](https://flax.readthedocs.io/en/latest/guides/initializers.html#initializer-restrictions-for-bias-init)`